drivers/usb/device/usb_dc_smartbond.c

```diff
@@ -890,36 +914,78 @@ static void usb_dc_smartbond_isr(void)
 	if (GET_BIT(int_status, USB_USB_MAEV_REG_USB_ALT)) {
 		handle_alt_ev();
 	}
+
+	for (int i = 0; dev_state.ep_out_data && i < 4; ++i) {
+		uint8_t mask = 1 << i;
+
+		if (dev_state.ep_out_data & mask) {
+			dev_state.ep_out_data ^= mask;
+			dev_state.ep_state[0][i].cb(dev_state.ep_state[0][i].ep_addr,
+						    USB_DC_EP_DATA_OUT);
+		}
+	}
 }
 
-static void usb_dc_smartbond_vbus_changed(bool present)
+/**
+ * USB functionality can be disabled from HOST and DEVICE side.
+ * Host side is indicated by VBUS line.
+ * Device side is decided by pair of calls usb_dc_attach()/usb_dc_detach,
+ * USB will only work when application calls usb_dc_attach() and VBUS is present.
+ * When both conditions are not met USB clock (PLL) is released, also PD_DOMAIN_SYS
+ * is released so device can use deep sleep.
+ */
+static void usb_state_check(void)
 {
-	if (dev_state.vbus_present == present) {
+	bool vbus_present =
+		(CRG_TOP->ANA_STATUS_REG & CRG_TOP_ANA_STATUS_REG_VBUS_AVAILABLE_Msk) != 0;
+
+	if (!dev_state.attached) {
 		return;
 	}
 
-	dev_state.vbus_present = present;
-
-	if (present) {
-		/* TODO: Add PD acquire when available */
-		/* If power event happened before USB started, delay dcd_connect
-		 * until dcd_init is called.
-		 */
+	if (!dev_state.vbus_present && vbus_present) {
+		dev_state.vbus_present = true;
+		da1469x_pd_acquire(MCU_PD_DOMAIN_SYS);
```

Comment:
We should better use PM constraints to totally prevent sleep. There is no benefit in initiating the PM process and end up entering the WFI state within `da1469x_enter_sleep` (because of the PD_SYS reference counter being > 0). Besides, when a USB device is attached, VBUS should become greater than VBAT and so the entire SoC will be powered by VBUS.

Reply:
@ioannis-karachalios instead of acquiring PD_SYS code now uses pm_policy_state_lock_put/get to prevent suspend state when PM is in use.